src/agentscope/formatter/_zhipu_formatter.py

```diff
@@ -0,0 +1,216 @@
+                        {
+                            "type": "text",
+                            "text": self.convert_tool_result_to_string(
+                                block.get("output"),  # type: ignore[arg-type]
+                            ),
+                        }
+                    )
+
+                elif typ == "image":
+                    source_type = block["source"]["type"]
+                    if source_type == "url":
+                        url = _to_zhipu_image_url(block["source"]["url"])
+                        content_blocks.append(
+                            {
+                                "type": "image_url",
+                                "image_url": {"url": url},
+                            },
+                        )
+                    elif source_type == "base64":
+                        data = block["source"]["data"]
+                        media_type = block["source"]["media_type"]
+                        url = f"data:{media_type};base64,{data}"
+                        content_blocks.append(
+                            {
+                                "type": "image_url",
+                                "image_url": {"url": url},
+                            },
+                        )
+
+                else:
+                    logger.warning(
+                        f"Unsupported message block type: {typ} in "
+                        f"ZhipuChatFormatter. Skipping this block.",
+                    )
+
+            message = {
+                "role": msg.role,
+            }
+
+            if (
+                len(content_blocks) == 1
+                and content_blocks[0]["type"] == "text"
+            ):
+                message["content"] = content_blocks[0]["text"]
+            elif content_blocks:
+                message["content"] = content_blocks
+
+            if tool_calls:
+                message["tool_calls"] = tool_calls
+
+            if "content" in message or tool_calls:
+                messages.append(message)
+
+        return messages
+
+
+class ZhipuMultiAgentFormatter(ZhipuChatFormatter):
+    """The class used to format message objects into the Zhipu AI API required
+    format for multi-agent conversation."""
+
+    async def _format(
+        self,
+        msgs: list[Msg],
+    ) -> list[dict[str, Any]]:
+        """Format message objects into Zhipu AI API required format for
+        multi-agent conversation.
+
+        Args:
+            msgs (`list[Msg]`):
+                The list of Msg objects to format.
+
+        Returns:
+            `list[dict[str, Any]]`:
+                A list of dictionaries, where each dictionary has "role",
+                "content", and optionally "name" keys.
+        """
+        messages = await super()._format(msgs)
+
+        for i, (msg, message) in enumerate(zip(msgs, messages)):
+            if msg.name and msg.name != msg.role:
+                message["name"] = msg.name
```

Comment:
Can the Zhipu API recognize the `name` attribute in messages? It's not mentioned in the official documentation, so this approach would cause issues in multi-agent conversations where a single agent cannot distinguish which agent said what. I suggest modifying this - you can see other multi-agent formatter functions, such as `OpenAIMultiAgentFormatter`.